### PR TITLE
fix Issue 19672 - Function uses the stack to read a static array para…

### PR DESCRIPTION
**Pull request by WalterBright**
…meter but it is passed in the registers

### CODE REVIEW

File: src/dmd/backend/cod1.d [hunk 1]
```diff
@@ -4678,7 +4685,7 @@ void pushParams(ref CodeBuilder cdb, elem* e, uint stackalign, tym_t tyf)
 
         default:
             break;
-   }
+    }
```
Comment (thewilsonator) [inline src/dmd/backend/cod1.d]: Eh?

Comment (thewilsonator) [reply src/dmd/backend/cod1.d]: Oh thats aligned now. nvm

**Review by thewilsonator — APPROVED** [review]

File: src/dmd/backend/cod1.d (hunk 1 above)
Comment (WalterBright) [reply src/dmd/backend/cod1.d]: A formatting error that creeped in on another PR. Not worth it's own PR.

**Comment by thewilsonator** [comment]
Please target stable. otherwise looks good to go.

**Comment by WalterBright** [comment]
I don't think it is an actual regression. Those checks have always been missing.

**Comment by ZombineDev** [comment]
@WalterBright all bug fixes should preferably go to stable, not only regressions.

**Comment by WalterBright** [comment]
> @WalterBright all bug fixes should preferably go to stable

This is not practical. Bug fixes can be destabilizing.

**Comment by ZombineDev** [comment]
That's what I meant by "preferably" - if they're not destabilizing.

**Comment by WalterBright** [comment]
AppVeyor says:
```
Error: unrecognized switch '-preview=fieldwise'
```
but this PR does not use that switch.

**Comment by ZombineDev** [comment]
I could agree that this particular bug fix is a bit more risky, so staying on naster is ok.

**Comment by wilzbach** [comment]
Looks like both Appveyor and buildkite fail to detect that the target branch is stable and incorrectly checkout phobos from master.

**Comment by wilzbach** [comment]
Merged as (1) there won't be a 2.084.2, but only 2.085 and (2) AppVeyor and Buildkite failure are unrelated (and everything else passed).

**Comment by WalterBright** [comment]
Thank you @wilzbach

**Comment by nordlow** [comment]
Why wasn't this merge into master?

**Comment by wilzbach** [comment]
Because it's a bug fix and __ALL BUGFIXES SHOULD GO TO STABLE__.

Also note that master will be merged into stable tomorrow anyhow (2.085 master/stable branch off is scheduled for 15.2).

**Comment by JinShil** [comment]
This PR may have introduced a regression: https://issues.dlang.org/show_bug.cgi?id=19804

**Comment by NilsLankila** [comment]
Another reg found too https://issues.dlang.org/show_bug.cgi?id=20890

**Pull request merged**